src/shogun/base/progress.h

```diff
@@ -58,6 +58,12 @@
 
 namespace shogun
 {
+#define SG_PROGRESS(...)                                                       \
+	progress(                                                                  \
+	    std::string(this->get_name()) + "::" + std::string(__FUNCTION__),      \
+	    *this->io, __VA_ARGS__)
+
+#define SG_SPROGRESS(...) progress("PROGRESS: ", __VA_ARGS__)
```

Comment:
I think just "PROGRESS" is the simplest version here :)

Reply:
Sorry for being picky here, but I realised that this is not nicest.
(This doesnt block a merge though, we can change this later)

 Maybe use the filename (without suffix) instead? @vigsterkr thoughts?

Reply:
Could use `__FILE__` and something like 
```
std::string remove_extension(const std::string& filename) {
    size_t lastdot = filename.find_last_of(".");
    if (lastdot == std::string::npos) return filename;
    return filename.substr(0, lastdot); 
}
```

Reply:
this is still an open question

Reply:
I would prefer to see the actual object name, because if I had three progress bar inside three different ``train()`` method, I would like to see to which model (``CPerceptron::train()``, ``CAveragedPerceptron::train()`` etc.) they belong. So it is okay as it is done now :)

Reply:
There is no object for SG_SPROGRESS. We can either make the caller of the macro pass something or use the __FUNCTION__. But I think just using "Progress" is not the best or?

Reply:
I have used the ```__FUNCTION__``` macro here now :)